packages/mdc-data-table/component.ts

```diff
@@ -80,6 +88,14 @@ export class MDCDataTable extends MDCComponent<MDCDataTableFoundation> {
     this.foundation_.setSelectedRowIds(rowIds);
   }
 
+  destroy() {
+    this.headerRow_.removeEventListener('change', this.handleHeaderRowCheckboxChange_);
+    this.content_.removeEventListener('change', this.handleRowCheckboxChange_);
+
+    this.headerRowCheckbox_.destroy();
```

Comment:
I think you need an `if (this.headerRowCheckbox_)` check

Reply:
As per current implementation `headerRowCheckbox_` is non-nullable and guaranteed to be initialized in `initialSyncWithDOM` method.